src/plans/stepsize.jl

```diff
@@ -229,72 +232,78 @@ Pass `:Messages` to a `debug=` to see `@info`s when these happen.
+The constructors return the functor to perform Armijo line search, where
+
+    (a::ArmijoLinesearch)(amp::AbstractManoptProblem, ams::AbstractManoptSolverState, i)
+
+of a [`AbstractManoptProblem`](@ref) `amp`, [`AbstractManoptSolverState`](@ref) `ams` and a current iterate `i`
+with keywords
+  * `p` (`a.candidate_point`) to pass memory for the candidate, which in the end also stores the resulting point,
+  * `η` (`-get_gradient(mp, get_iterate(s));`) the search direction to use, by default the steepest descent direction.
 """
-mutable struct ArmijoLinesearch{TRM<:AbstractRetractionMethod,F} <: Linesearch
+mutable struct ArmijoLinesearch{TRM<:AbstractRetractionMethod,P,F} <: Linesearch
+    candidate_point::P
+    contraction_factor::Float64
+    initial_guess::F
     initial_stepsize::Float64
+    last_stepsize::Float64
+    message::String
     retraction_method::TRM
-    contraction_factor::Float64
     sufficient_decrease::Float64
-    last_stepsize::Float64
     stop_when_stepsize_less::Float64
     stop_when_stepsize_exceeds::Float64
     stop_increasing_at_step::Int
     stop_decreasing_at_step::Int
-    initial_guess::F
-    message::String
     function ArmijoLinesearch(
         M::AbstractManifold=DefaultManifold();
-        initial_stepsize::Float64=1.0,
-        retraction_method::AbstractRetractionMethod=default_retraction_method(M),
-        contraction_factor::Float64=0.95,
-        sufficient_decrease::Float64=0.1,
-        stop_when_stepsize_less=0.0,
-        stop_when_stepsize_exceeds=max_stepsize(M),
-        stop_increasing_at_step=100,
-        stop_decreasing_at_step=1000,
+        candidate_point::P=allocate_result(M, rand),
+        contraction_factor::Real=0.95,
+        initial_stepsize::Real=1.0,
         initial_guess=armijo_initial_guess,
-    )
-        return new{typeof(retraction_method),typeof(initial_guess)}(
+        retraction_method::TRM=default_retraction_method(M),
+        stop_when_stepsize_less::Real=0.0,
+        stop_when_stepsize_exceeds::Real=max_stepsize(M),
+        stop_increasing_at_step::Int=100,
+        stop_decreasing_at_step::Int=1000,
+        sufficient_decrease=0.1,
+    ) where {TRM,P}
+        return new{TRM,P,typeof(initial_guess)}(
+            candidate_point,
+            contraction_factor,
+            initial_guess,
+            initial_stepsize,
             initial_stepsize,
+            "",                                           # initilize an empty message
             retraction_method,
-            contraction_factor,
             sufficient_decrease,
-            initial_stepsize,
             stop_when_stepsize_less,
             stop_when_stepsize_exceeds,
             stop_increasing_at_step,
             stop_decreasing_at_step,
-            initial_guess,
-            "",
         )
     end
 end
 function (a::ArmijoLinesearch)(
     mp::AbstractManoptProblem,
     s::AbstractManoptSolverState,
     i::Int,
+    p=a.candidate_point,
```

Comment:
This `p` isn't used anywhere in this function.

Reply:
that might indeed be the bug, will check, since before the fourth was always only the search direction to use. Gradient uses the default CG does not, so this might be a point here.

Reply:
Super well spotted, that seems to fix it and I even have no clue how that got there ;)

Reply:
:tada: